packages/frontity/src/steps/create/index.ts

```diff
@@ -1,80 +1,213 @@
+    if (key === "settings") return JSON.stringify(frontitySettings, null, 2);
   });
+  await writeFile(filePath, fileData);
+};
+
+// This functions clones the starter theme.
+export const cloneStarterTheme = async ({ theme, path }: Options) => {
+  const packageJsonPath = resolvePath(path, "./package.json");
+  const packageJson = JSON.parse(
+    await readFile(packageJsonPath, { encoding: "utf8" })
+  );
+  const themePath = resolvePath(path, packageJson.dependencies[theme]);
+  await ensureDir(themePath);
+  if (!isThemeNameValid(theme))
+    throw new Error("The name of the theme is not a valid npm package name.");
+  await promisify(exec)(`npm pack ${theme}`, { cwd: themePath });
+  const tarball = (await readDir(themePath)).find(file => /\.tgz$/.test(file));
+  const tarballPath = resolvePath(themePath, tarball);
+  await extract({ cwd: themePath, file: tarballPath, strip: 1 });
+  await remove(tarballPath);
+};
+
+// This function installs the Frontity packages.
+export const installDependencies = async ({ path }: Options) => {
+  await promisify(exec)("npm install", { cwd: path });
+};
+
+// This function downlaods the favicon file.
+export const downloadFavicon = async ({ path }: Options) => {
+  const response = await fetch(faviconUrl);
+  const fileStream = createWriteStream(resolvePath(path, "favicon.ico"));
+  response.body.pipe(fileStream);
+  await new Promise(resolve => fileStream.on("finish", resolve));
+};
 
-  try {
-    // 1. Parses and validates options.
-    options = normalizeOptions(defaultOptions, passedOptions);
-
-    // 2. Ensures that the project dir exists and is empty.
-    step = ensureProjectDir(options);
-    emit(`Ensuring ${chalk.yellow(options.path)} directory.`, step);
-    dirExisted = await step;
-
-    // 3. Creates `package.json`.
-    step = createPackageJson(options);
-    emit(`Creating ${chalk.yellow("package.json")}.`, step);
-    await step;
-
-    // 4. Creates `frontity.settings`.
-    const extension = options.typescript ? "ts" : "js";
-    step = createFrontitySettings(extension, options);
-    emit(`Creating ${chalk.yellow(`frontity.settings.${extension}`)}.`, step);
-    await step;
-
-    // 5. Clones `@frontity/mars-theme` inside `packages`.
-    step = cloneStarterTheme(options);
-    emit(`Cloning ${chalk.green(options.theme)}.`, step);
-    await step;
-
-    // 6. Installs dependencies.
-    step = installDependencies(options);
-    emit(`Installing dependencies.`, step);
-    await step;
-
-    // 7. Download favicon.
-    step = downloadFavicon(options);
-    emit(`Downloading ${chalk.yellow("favicon.ico")}.`, step);
-    await step;
-  } catch (error) {
-    if (typeof dirExisted !== "undefined")
-      await revertProgress(dirExisted, options);
-
-    if (emitter) emitter.emit("error", error);
-    else throw error;
+// This function removes the files and directories created
+// with `frontity create`.
+export const revertProgress = async (
```

Comment:
I'd rename this step so it can be reused easily. Something like `removeDir` for example. And it could receive an optional array of `allowedExistingContent` passed by the command.